src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -997,6 +997,11 @@ public override int GetHashCode()
             return GetLegacyNonRandomizedHashCode();
         }
 
+        // Gets a hash code for this string and this comparison. If strings A and B and comparition C are such
+        // that String.Equals(A, B, C), then they will return the same hash code with this comparison C.
+        [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
```

Comment:
@jkotas is ReliabilityContract meaningful in Core? If not I will open an issue to remove it.